### ISSUE TITLE
Error message overflows window

### ISSUE THREAD

**Issue by DougInAMug** [borgbase/vorta#1974]
### Description

A failed backup lead to an error message long enough to overflow the window. No scroll bar was available, so reading was only possible by dragging window partially off the top of the screen, then enlarging the window.

![Screenshot from 2024-03-21 14-24-56](https://github.com/borgbase/vorta/assets/13798767/a9532887-c7b1-40b8-b18b-8682f1705c8f)

### Reproduction

- [X] I tried to reproduce the issue.
- [X] I was able to reproduce the issue.

### OS

Pop!_OS 22.04 LTS, GNOME 42.9

### Version of Vorta

0.8.3

### What did you install Vorta with?

Distribution package

### Version of Borg

1.2.0

**Comment by SAMAD101** [comment]
> Yes, could be solved better. And we already do some output parsing and could catch this and treat it differently.
> 
> We should first see how we want to solve it. Maybe as larger window, similar to Vorta's own exceptions.

Vorta's own exceptions could serve well. But what would be the best ways to catch exceptions outside of the Vorta's main thread? 

This could be a good addition to my projects ('Error handling & report wizard' one).
What do you think?

**Comment by shivansh02** [comment]
> We should first see how we want to solve it. Maybe as larger window, similar to Vorta's own exceptions.
We can show a few lines and add a Read More button, which would open a scrollable window with all the details.

**Comment by AdwaitSalankar** [comment]
I think it's better to have a pop-up window with a copy to clipboard button.

**Comment by shivansh02** [comment]
@m3nu Should I remove the error from here and just open an exception box as done in #1977?

**Comment by m3nu** [comment]
Yes, I would use the same error box for Vorta AND Borg exceptions. Even if their content will be a bit different and the source they come from.

**Comment by shivansh02** [comment]
Sure, will take this up as soon as I'm done with my proposal submission.

**Comment by ahmedhosssam** [comment]
Hi, I'm working on this issue right now.

**Comment by m3nu** [comment]
Hi, sorry for the delay. I just got back to this. I think we have 2 tasks here:

1. Better handle real errors (done in this PR by @ahmedhosssam)
2. Better handle normal logs that are slightly too long. By maybe truncating them. So the Eg.

<img width="807" height="132" alt="Screenshot 2025-07-13 at 07 08 05" src="https://github.com/user-attachments/assets/45b3055e-5a67-4711-980f-17bffd3cf9f3" />

**Comment by DougInAMug** [comment]
Nice one @m3nu ! ✨
